### DISCUSSION TITLE
Unable to listen to the events of any child component from a parent

### DISCUSSION THREAD

**Discussion by ghost** [alpinejs/alpine#1769] [7. Bugs (archived)]
Hello,

When we bind listener from parent component to child component, there is no way to change x-data of the parent using that listener, it is because the listener function will be bound to the child components x-data object.

Please check the live example here: https://codepen.io/sharifzadesina/pen/zYwKgNW

**Comment by Pronian** [comment]
Your `@trigger="myListener"` is added to the child component, if you move it to the parent, it will work. I believe it's cleaner and more concise to have this data separation between parent and child.

**Reply by ghost** [reply]
The whole point is to have it on child!!!
It is for communication.

**Reply by Pronian** [reply]
Why is this necessary? What is the use case?

I think it's confusing to have the child communicate with "itself" using different data.

**Reply by ghost** [reply]
Bro, child isn't communicating with itself, child dispatches an event, and parent wants to listen to that event and store data of that event to a data property.
And this use case is fairly common.

**Reply by Pronian** [reply]
But that's not what is happening in the example you provided.

In your example the child the dispatchers the event and listens for the event, the parent is not involved in the communication.

**Reply by ghost** [reply]
The listener is from parent, not child.

**Reply by Pronian** [reply]
Your `@trigger="myListener"` code is set on the second x-data component, inside the first one. The first one is the parent and the second one is the child. The listener is added to the second one, but not the first.

**Reply by ghost** [reply]
Yes, the listener is added from parent to child.
Like Vue or any other frameworks.
